shop/src/components/payment-widget.js

```diff
@@ -42,39 +41,19 @@ async function getDefaultAccountAddr() {
  */
 async function executePayment(fromAddress, toAddress, amount, setLoading) {
   try {
-    // make transaction
-    const txParams = await AlgoSigner.algod({
-      ledger: CHAIN_NAME,
-      path: '/v2/transactions/params',
-    });
-    const txn = {
-      fee: txParams['min-fee'],
-      firstRound: txParams[LAST_ROUND],
-      lastRound: txParams[LAST_ROUND] + 1000,
-      genesisHash: txParams['genesis-hash'],
-      genesisID: txParams['genesis-id'],
-      from: fromAddress,
-      type: 'pay',
-      amount: amount,
-      to: toAddress,
+    const tx = new WebMode(AlgoSigner, CHAIN_NAME);
+    const txParams = {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccountAddr: fromAddress,
+      toAccountAddr: toAddress,
+      amountMicroAlgos: amount,
+      payFlags: {},
     };
-
-    // trigger signing prompt
-    let signedTxn = await AlgoSigner.sign(txn);
-
-    // show loading state on button while we send & wait for transaction response
-    setLoading(true);
```

Comment:
Could you please set this before `tx.executeTransaction(txParams);` ? We need to show a loading state at UI while transaction is being processed.

Reply:
ok